components/camel-mina2/src/main/java/org/apache/camel/component/mina2/Mina2Producer.java

```diff
@@ -502,6 +511,7 @@ public void sessionClosed(IoSession session) throws Exception {
                 // and could not return a response. We should count down to stop waiting for a response
                 countDown();
             }
+            closeLatch.countDown();
```

Comment:
What if the session is closed remotely before the producer is able to close it? Won't closeLatch be uninitialised?